db/write_callback_test.cc

```diff
@@ -145,23 +145,29 @@ TEST_F(WriteCallbackTest, WriteWithCallbackTest) {
               db_impl = dynamic_cast<DBImpl*>(db);
               ASSERT_TRUE(db_impl);
 
+              std::atomic<uint64_t> threads_joined(0);
+              std::atomic<uint64_t> threads_linked(0);
               std::atomic<uint64_t> threads_waiting(0);
               std::atomic<uint64_t> seq(db_impl->GetLatestSequenceNumber());
               ASSERT_EQ(db_impl->GetLatestSequenceNumber(), 0);
 
               rocksdb::SyncPoint::GetInstance()->SetCallBack(
+                  "WriteThread::JoinBatchGroup:Start", [&](void* arg) {
+                    uint64_t cur_threads_joined = threads_joined.fetch_add(1);
+                    // Wait for threads joined earlier to finish linking.
```

Comment:
This is a callback for Start. If everybody sync here, nobody has reached Wait yet. So Waiting for threaded who joined earlier (using the callback on Wait) does not make sense.